### ISSUE TITLE
Loss Functions and Optimizers

### ISSUE THREAD

**Issue by Shaswat-G** [Shaswat-G/PyTorch#7]
Cover the most commonly used loss functions and optimizers with all their implementation details.
